packages/wdio-cli/src/utils.js

```diff
@@ -17,6 +17,44 @@ export async function runServiceHook (launcher, hookName, ...args) {
     }
 }
 
+/**
+ * Run onPrepareHook in Launcher
+ * @param {Array|Function} onPrepareHook - can be array of functions or single function
+ * @param {Object} config
+ * @param {Object} capabilities
+ */
+export async function runOnPrepareHook(onPrepareHook, config, capabilities) {
```

Comment:
One issue with this .. this is not backwards compatible. In order to not break every test we should ensure that functions are moved into an array.

Reply:
The onPrepare and onComplete functions are added to the root config array in the ConfigParser.addFileConfig() -> addService() function now.

Reply:
These should be backwards compatible regardless of the ConfigParser change.